### DISCUSSION TITLE
Dataframe View

### DISCUSSION THREAD

**Discussion by SeeRich** [hosseinmoein/DataFrame#200] [General]
This library looks amazing, great work!

Is there any support for a "DatafFrame View", i.e. a DataFrame that uses a non-owning reference to the column and index data?

**Comment by hosseinmoein** [comment]
Yes, if you read the docs here https://hosseinmoein.github.io/DataFrame/docs/HTML/DataFrame.html there is the concept of views which are refernces to the original DataFrames. For all slicing operations, you can choose to get a view

**Reply by SeeRich** [reply]
Thank you, I just looked at the referenced documentation. The only downside to that method is that it requires me to already have an existing DataFrame. My situation is that I already have a list of vectors of pod data that I would love to use the visitors in this library on. However, I cannot move those vectors into the DataFrame (I.e. giving up ownership). I guess I was hoping/looking for something that could take a const reference to the vectors and would enable any const methods of the DataFrame class + visitors. Maybe something like ‘ConstDataFrame’?

**Reply by hosseinmoein** [reply]
Visitors are designed for DataFrame, but why can’t you move your data vectors to DataFrame and use it as your main place for data? You can still easily get references to your vectors

**Reply by SeeRich** [reply]
I am currently trying to fit the usage of DataFrame into my existing architecture. It is a tree based processing application where each node in the tree receives a read-only/const block of data (the block is currently an unordered_map of vectors mapped on their name). If the node needs to modify the data, it makes a copy, modifies and passes on that block for the child nodes (which would use as read only). The benefit of this are the cache hits for nodes that only need read access to the block (they don't need to make a copy). If each node moved the existing vectors into a DataFrame, it would invalidate the block for other readers.
